plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools.tsx

```diff
@@ -0,0 +1,185 @@
+import { recordEvent } from '@woocommerce/tracks';
+import { Pill, EmptyContent, Spinner } from '@woocommerce/components';
+import { flatMapDeep, uniqBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CollapsibleCard,
+	CardDivider,
+	CardBody,
+	PluginCardBody,
+} from '~/marketing/components';
+import { STORE_KEY } from '~/marketing/data/constants';
+import { getInAppPurchaseUrl } from '~/lib/in-app-purchase';
+import './DiscoverTools.scss';
+
+const category = 'marketing';
+
+type SubcategoryType = {
+	slug: string;
+	name: string;
+};
+
+type TagType = {
+	slug: string;
+	name: string;
+};
+
+type Plugin = {
+	title: string;
+	description: string;
+	url: string;
+	icon: string;
+	product: string;
+	plugin: string;
+	categories: Array< string >;
+	subcategories: Array< SubcategoryType >;
+	tags?: Array< TagType >;
+};
+
+type SelectResult = {
+	isLoading: boolean;
+	plugins: Plugin[];
+};
+
+/**
+ * Return tabs (`{ name, title }`) for the TabPanel.
+ *
+ * Subcategories that have no plugins
+ * will not be displayed as a tab in the UI.
+ * This is done by doing the following:
+ *
+ * 1. Get an array of unique subcategories from the list of plugins.
+ * 2. Map the subcategories schema into tabs schema.
+ */
+const getTabs = ( plugins: Plugin[] ) => {
+	const pluginSubcategories = uniqBy(
+		flatMapDeep( plugins, ( p ) => p.subcategories ),
+		( subcategory ) => subcategory.slug
+	);
+
+	return pluginSubcategories.map( ( subcategory ) => ( {
+		name: subcategory.slug,
+		title: subcategory.name,
+	} ) );
+};
+
+const renderPluginCardBodies = ( plugins: Plugin[] ) => {
+	return plugins.map( ( el, idx ) => {
+		return (
+			<Fragment key={ el.product }>
+				<PluginCardBody
+					icon={ <img src={ el.icon } alt={ el.title } /> }
+					name={ el.title }
+					pills={ el.tags?.map( ( tag ) => (
+						<Pill key={ tag.slug }>{ tag.name }</Pill>
+					) ) }
+					description={ el.description }
+					button={
+						<Button
+							variant="secondary"
+							href={ getInAppPurchaseUrl( el.url ) }
+							onClick={ () => {
+								recordEvent(
+									'marketing_recommended_extension',
+									{ name: el.title }
+								);
+							} }
+						>
+							{ __( 'Get started', 'woocommerce' ) }
+						</Button>
+					}
+				/>
+				{ idx !== plugins.length - 1 && <CardDivider /> }
+			</Fragment>
+		);
+	} );
+};
+
+export const DiscoverTools = () => {
+	const { isLoading, plugins } = useSelect< SelectResult >(
+		( select ) => {
+			const { getRecommendedPlugins, isResolving } = select( STORE_KEY );
+
+			return {
+				isLoading: isResolving( 'getRecommendedPlugins', [ category ] ),
+				plugins: getRecommendedPlugins( category ),
+			};
+		},
+		[ category ]
+	);
+
+	/**
+	 * Renders card body.
+	 *
+	 * - If loading is in progress, it renders a loading indicator.
+	 * - If there are zero plugins, it renders an empty content.
+	 * - If the plugins do not have subcategories field, it renders the list of plugins without TabPanel.
```

Comment:
❓  I tested this by removing 'subcategories' field and I got a fatal

<img width="870" alt="Screenshot 2022-08-23 at 15 44 42" src="https://user-images.githubusercontent.com/5908855/186149742-f847b5b2-ef3e-43ed-aa9d-dcd47348990c.png">

Reply:
@puntope , thanks for testing that scenario. The code and comment are now outdated, as we are not using the new recommendations API version 1.2 and subcategories and tags are there in the API. I have updated the code and comment in [`bc835ac` (#34415)](https://github.com/woocommerce/woocommerce/pull/34415/commits/bc835ac71d8e8d93dc4599d0f242e97e35a6c961).